### DISCUSSION TITLE
Navigation with dispatched action

### DISCUSSION THREAD

**Discussion by feco911** [mrpmorris/Fluxor#513] [Q&A]
Hi,

We have run into a scenario, when first an action is being dispatched, then a navigation occurs using the `NavigationManager`. The dispatched action triggers an effect.

Are we guaranteed the action/effect has run completely before the navigation?

Scenario:
Dispatcher.Dispatch(new XAction()); // this triggers an effect 
NavigationManager.NavigateTo(....);

Thanks in advance

**Comment by mrpmorris** [comment]
You are guaranteed that the action has been reduced into the states, but there is no guarantee as to which order effects will be executed so the navigation could happen first or last.
